### PR TITLE
System.IO.Pipelines readonly annotation

### PR DESCRIPTION
**Pull request by hrrrrustic**
This PR is a part of #46675

### CODE REVIEW

File: src/libraries/System.IO.Pipelines/src/System/IO/Pipelines/PipeCompletion.cs [hunk 1]
```diff
@@ -88,7 +88,7 @@ public void Reset()
             _state = null;
```
Comment (Wraith2) [inline src/libraries/System.IO.Pipelines/src/System/IO/Pipelines/PipeCompletion.cs:88]: this assignment to `_state` means that readonly can't be used here.

File: src/libraries/System.IO.Pipelines/src/System/IO/Pipelines/PipeAwaitable.cs [hunk 2]
```diff
@@ -19,7 +19,7 @@ internal struct PipeAwaitable
         private CancellationTokenRegistration _cancellationTokenRegistration;
 
 #if (!NETSTANDARD2_0 && !NETFRAMEWORK)
-        private CancellationToken CancellationToken => _cancellationTokenRegistration.Token;
+        private readonly CancellationToken CancellationToken => _cancellationTokenRegistration.Token;
 #else
         private CancellationToken _cancellationToken;
         private CancellationToken CancellationToken => _cancellationToken;
```
Comment (Wraith2) [inline src/libraries/System.IO.Pipelines/src/System/IO/Pipelines/PipeAwaitable.cs:25]: If line 22 can be readonly can't 25 also be readonly? they're the same property in different builds.

File: src/libraries/System.IO.Pipelines/src/System/IO/Pipelines/PipeCompletion.cs (hunk 1 above)
Comment (hrrrrustic) [reply src/libraries/System.IO.Pipelines/src/System/IO/Pipelines/PipeCompletion.cs:88]: image
There is no ```readonly``` at ```Reset()```. Github diff merged ```IsCompletedOrThrow()``` and ```Reset()``` :)

Comment (Wraith2) [reply src/libraries/System.IO.Pipelines/src/System/IO/Pipelines/PipeCompletion.cs:88]: Ah, my bad, unhelpful diff window size.

**Comment by jeffhandley** [comment]
This PR is being marked as a draft per https://github.com/dotnet/runtime/pull/46675#issuecomment-828694665.

**Pull request closed without merging**